db/write_callback_test.cc

```diff
@@ -144,24 +148,37 @@ TEST_F(WriteCallbackTest, WriteWithCallbackTest) {
 
               db_impl = dynamic_cast<DBImpl*>(db);
               ASSERT_TRUE(db_impl);
-
+              
+              // Writers that have called JoinBatchGroup.
+              std::atomic<uint64_t> threads_joined(0);
+              // Writers that have linked to the queue
+              std::atomic<uint64_t> threads_linked(0);
+              // Writers that passes WriteThread::JoinBatchGroup:Wait sync-point.
               std::atomic<uint64_t> threads_waiting(0);
+
               std::atomic<uint64_t> seq(db_impl->GetLatestSequenceNumber());
               ASSERT_EQ(db_impl->GetLatestSequenceNumber(), 0);
 
               rocksdb::SyncPoint::GetInstance()->SetCallBack(
+                  "WriteThread::JoinBatchGroup:Start", [&](void* arg) {
+                    uint64_t cur_threads_joined = threads_joined.fetch_add(1);
+                    // Wait for other writers linking to the queue. In this way
+                    // we will know whether the writer is a leader by checking
```

Comment:
What is we do not do this? Would not threads_linked == 0 indicate that a thread is the leader?